elastic/eui
# [Storybook] update Storybook utils to ensure merging of control configs

## Summary

This PR updates the internal Storybook utils `hideStorybookControls`, `disableStorybookControls` and `moveStorybookControlsToCategory` to ensure they merge control table updates into the Preview/Component/Story config instead of potentially overwriting existing settings.

To ensure this, the functions now take the config as an argument and add the new control setting onto it. 
This means the following things:
- the util should be called after creating the config
- the passed config is mutated (limitation of current story CSF setup)
- multiple utils can be called without overwriting other control settings unless the same key is overwritten with a primitive value

The proposed way of using the utils:

```js
const meta: Meta<EuiComponentProps> = {
  title: 'EuiComponentName',
  component: EuiComponent,
  argTypes: {
    ...
  },
  args: {
    ...
  },
};

moveStorybookControlsToCategory(meta, ['controlName']);
disableStorybookControls(meta, ['controlName']);
```

Additionally, this PR updates all usages of those three util functions inside available story files. 

## QA

- [x] check an example story (e.g. for [SkipLink](https://eui.elastic.co/pr_7611/storybook/index.html?path=/story/utilities-euiskiplink--playground)) and verify that:
    - [x] the props `color`, `fill` and `size` are moved to the category `EuiButton props`
- [x] checkout this pr locally and verify that multiple calls of utils on the same config work as expected
    - `gh pr checkout 7611`
    - open `facet_button_stories.tsx`
    - add `fullWidth` to `disableStorybookControls(meta, ['buttonRef']);` on line 56
    - verify that the control is moved to the category `EuiButtonEmpty props` AND it's now disabled instead of type `boolean`
- [x] verify that the props `css`, `className` and `data-test-subj` are not displayed on any story control table (globally hidden in `preview.tsx`)
